lib/workers/global/limits.ts

```diff
@@ -15,15 +16,17 @@ export function resetAllLimits(): void {
   limits.clear();
 }
 
-export function setMaxLimit(key: Limit, max: unknown): void {
-  const maxVal = typeof max === 'number' && max > 0 ? max : null;
-  logger.debug(`${key} limit = ${maxVal}`);
-  const limit = limits.get(key);
-  limits.set(key, {
-    current: 0,
-    ...limit,
-    max: maxVal,
-  });
+export function setMaxLimit(key: Limit, max: unknown, allowZero = false): void {
+  if (typeof max === 'number') {
+    const maxVal = max > 0 || (allowZero && max <= 0) ? Math.max(max, 0) : null;
+    logger.debug(`${key} limit = ${maxVal}`);
+    const limit = limits.get(key);
+    limits.set(key, {
+      current: 0,
+      ...limit,
+      max: maxVal,
+    });
+  }
```

Comment:
Shouldn't this change be a seperate pr?

Reply:
Maybe, just thought it's not so substantial and we already have too much preparatory PRs for branch concurrent limits